### PR TITLE
Tidy shell scripts in ./bin

### PR DESCRIPTION
**Pull request by IsaacG**
* Prefer `[[` over `[`
* Expand tabs to 4 spaces
* Quote variables uniformly in `"${var}"` format
* Use `local` for function variables
* Prefer lowercase variable names for non-readonly variables
* Replace `let` with `$((`

### CODE REVIEW

File: bin/check_required_files_present [hunk 1]
```diff
@@ -1,64 +1,64 @@
...
 
-if [ ! -d exercises ] ; then
-	echo "\"exercises\" directory not found, make sure you're running from the repository root" >&2
-	exit 1
+if [[ ! -d exercises ]] ; then
+    echo '"exercises" directory not found, make sure you are running from the repository root' >&2
+    exit 1
 fi
 
 function show_progress {
```
Comment (glennj) [inline bin/check_required_files_present:10]: Not comfortable with a `show_progress` function and a `show_progress` variable. Rename the variable?

File: bin/check_required_files_present [hunk 2]
```diff
@@ -1,64 +1,64 @@
...
+            check_directory 'exercises'
+            if [[ "${show_progress}" == 'true' ]]; then
+                echo
+                echo "Done: ${missing_files} files missing."
+            fi
+            if (( missing_files > 0 )); then
+                exit 1
+            else
+                exit 0
+            fi
```
Comment (glennj) [inline bin/check_required_files_present:60]: Is `exit "$(( missing_files == 0 ))"` too obscure?

**Review by glennj — CHANGES_REQUESTED** [review]

File: bin/check_required_files_present (hunk 2 above)
Comment (IsaacG) [reply bin/check_required_files_present:60]: I'd prefer `(( missing_files == 0 )); exit` ... but being verbose here is probably good. It's longer but less likely to give anyone pause.

File: bin/check_required_files_present (hunk 1 above)
Comment (IsaacG) [reply bin/check_required_files_present:10]: I'll wait on the other PRs and rebasing.

**Review by ErikSchierboom — APPROVED** [review]

File: bin/check_required_files_present (hunk 1 above)
Comment (IsaacG) [reply bin/check_required_files_present]: Renamed.

**Review by glennj — APPROVED** [review]

File: bin/reorder-keys [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+        walk(
+            if (type == "object") and has("uuid") then
+                . as $obj |
+            reduce ["uuid", "reimplements", "description", "comments", "scenarios", "property", "input", "expected"][] as $key (
+                {};
+                . + (if $obj | has($key) then { ($key): $obj[$key] } else {} end)
+            )
+            else
+                .
+            end
```
[the remark this answers is not in the document]
Comment (IsaacG) [reply bin/reorder-keys:21]: Yes. There were tabs in the file which messed some things up. Fixing!

**Review by kytrinyx — APPROVED** [review]

**Pull request merged**